team-cost-reporter/aws_cost_collector.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import datetime
+import boto3
+
+logging.getLogger('botocore').setLevel(logging.CRITICAL)
+
+
+def id():
+    return "aws_cost_collector"
+
+
+def log (message):
+    print id() + ": " + message
+
+
+def get_start_end_strings(number_of_days):
+    ''' get the start and end date strings based on the number of days to report '''
+    now = datetime.datetime.now()
+    n_days_ago = now - datetime.timedelta(days=number_of_days)
+
+    current_date_str = now.strftime("%Y-%m-%d")
+    n_days_ago_str = n_days_ago.strftime("%Y-%m-%d")
+
+    # start=2016-06-01
+    # end=2016-06-30
+    start_string = n_days_ago_str
+    end_string = current_date_str
+    return (start_string,end_string)
+
+
+def get_costs(days_to_report, granularity, filter, group_by, next_token=None, debug=False):
+    SESSION = boto3.session.Session()
```

Comment:
Actually, I'm not crazy about this - I think I'll move this outside this function - doesn't make sense to be inside it.